feat: create notes using HackMD

mmarchini opened the pull request.

mmarchini commented:
Passing CI on my fork: https://github.com/mmarchini/meet/runs/964840886?check_suite_focus=true
Issue generated by test: https://github.com/mmarchini/meet/issues/14

mmarchini commented:
cc @wesleytodd :)

wesleytodd commented on lib/notes.js:
```diff
@@ -0,0 +1,25 @@
+  const resp = await client.repos.getContents({
+    owner: opts.owner,
+    repo: opts.repo,
+    path: `.github/ISSUE_TEMPLATE/${opts.notesTemplate || 'meeting-notes.md'}`
```
I think we should move this to `.github/` right?  Since it is technically not an issue template?

bnb replied:
I would say we may want to have our own folder in `.github/` - perhaps `.github/meet/`. But yes, I agree with this.

wesleytodd commented on run.js:
```diff
@@ -14,8 +16,10 @@ const defaultTemplate = require('./lib/default-template')
     const issueTemplate = core.getInput('issueTemplate')
     const createWithin = core.getInput('createWithin')
-    const agendaLabel = core.getInput('agendaLabel')
+    const agendaLabel = core.getInput('agendaLabel') || 'meeting-agenda'
```
I think you can set this as a default for the action instead of here, right? The benefit being it would show in the action docs.

wesleytodd replied:
Hm, I see it is set as a default above.  Are both necessary?

mmarchini replied:
No idea, this was my first time working on an actual Action (instead of consuming one), so I wanted to be extra safe 😅. It's probably not necessary though.

bnb replied:
Yep, setting it as the default would be ideal.

wesleytodd commented:
cc: @pkgjs/meet-collaborators

bnb commented on lib/default-notes-template.js:
```diff
@@ -0,0 +1,34 @@
+
+  *
+
+## Agenda
+
+## Announcements
```
Are these flipped? Agenda items are below announcements.

mmarchini replied:
On all wgs I participate "Announcements" appear after "Agenda", but I'm happy to flip (it does make more sense to flip)

bnb replied:
That may be a copy/paste template issue - I'd recommend we flip 😅

bnb commented on lib/default-notes-template.js:
```diff
@@ -0,0 +1,34 @@
+
+## Upcoming Meetings
+
+* **Node.js Foundation Calendar**: https://nodejs.org/calendar
```
It would be nice if we could make this agnostic of Node.js.

mmarchini replied:
The template can be changed, but we could make this an Action parameter (doesn't have to be in this PR though).

bnb replied:
I'm good with removing now and making an Action parameter (this is how I initially set it up in my version of the same project, IIRC) or removing and PR'ing later.

bnb reviewed (approved):
Pending comments, huge +1. Thank you for this work, Mary ❤️

boneskull commented:
if I can be of any help getting this to land (or developing it further), please let me know!

- maybe a template engine like handlebars or liquid would be helpful
- also interested in figuring out how to tag stuff, as I know HackMD has added some API endpoints we could use
- can work on adoption by OJSF CPC

bnb commented:
+1 @boneskull

bnb commented:
@mmarchini any update on this? ❤️

mmarchini commented:
Addressed all inline comments and rebased.

> maybe a template engine like handlebars or liquid would be helpful

I _think_ that's already how it works, but if it's not we should do it in a separate PR and include issue templating as well

> also interested in figuring out how to tag stuff, as I know HackMD has added some API endpoints we could use

That sounds interesting, let's open an issue to investigate in the future

> can work on adoption by OJSF CPC

<3

mmarchini commented:
Test on my fork: https://github.com/mmarchini/meet/pull/15/checks?check_run_id=1387198907
Issue created: https://github.com/mmarchini/meet/issues/17

Seems like I messed something up and now the issue template is not being used 🤔

mmarchini commented:
Ok, should be working now.

Test on my fork: https://github.com/mmarchini/meet/runs/1387488797?check_suite_focus=true
Issue created: https://github.com/mmarchini/meet/issues/34
Notes created: https://hackmd.io/bAoXtmZ2QRKlFmXDR9LBMQ
(notes aren't showing on the issue because the Action will get the template directly from the default branch, which doesn't have `meetingNotes` yet, should work once we merge)

bnb commented on lib/meetings.js:
```diff
@@ -11,10 +11,37 @@ module.exports.createNextMeeting = async function (client, opts) {
+    date,
+    agendaLabel: opts.agendaLabel,
+    agendaIssues: opts.agendaIssues,
+    labels: opts.labels,
```
This will need to change with https://github.com/pkgjs/meet/pull/39
